pkg/northbound/roles/roles_test.go

```diff
@@ -153,3 +154,42 @@ func Test_BadSet(t *testing.T) {
 	})
 	assert.Assert(t, err != nil, "unable to issue set delete request")
 }
+
+func Test_Subscribe(t *testing.T) {
+	conn, client := getClient()
+	defer conn.Close()
+	stream, err := client.Subscribe(context.Background(), &proto.DeviceRoleRequest{})
+	assert.NilError(t, err, "unable to issue subscribe request")
+
+	go func() {
+		time.Sleep(100 * time.Millisecond)
+		conn1, client1 := getClient()
+		defer conn1.Close()
+		testSetRole(t, client1, &proto.DeviceRoleChangeRequest{
+			Change: proto.DeviceRoleChangeRequest_ADD,
+			Config: &spineRole,
+		})
+	}()
+
+	w := sync.WaitGroup{}
```

Comment:
Nothing wrong with this code, but FYI I've found the `sync` primitives to be especially prone to the copy issue. Create a `sync.WaitGroup{}` and then pass it by value to a function, and you'll be waiting forever. Conversely, in this case it doesn't seem to be copied into the `go func()`, right? But change to:
```go
go func(wg sync.WaitGroup) {
    ...
}()
```
and I bet you'll have problems.

Reply:
Yes, it seems like the new set of linters catches these sorts of things.